tests/test.py

```diff
@@ -4497,6 +4498,152 @@ def get_poison_record_data(self):
+        self.test_table = sa.Table(
+            'test_tokenization_default_client_id', self.get_metadata(),
+            sa.Column('id', sa.Integer, primary_key=True),
+            sa.Column('nullable', sa.Text, nullable=True),
+            sa.Column('empty', sa.LargeBinary(length=base.COLUMN_DATA_SIZE), nullable=False, default=b''),
+            sa.Column('token_i32', sa.Integer()),
+            sa.Column('token_i64', sa.BigInteger()),
+            sa.Column('token_str', sa.Text),
+            sa.Column('token_bytes', sa.LargeBinary(length=base.COLUMN_DATA_SIZE), nullable=False, default=b''),
+            sa.Column('token_email', sa.Text),
+        )
+        self.get_metadata().create_all(self.engine_raw, [self.test_table])
+
+    def tearDown(self):
+        if hasattr(self, 'test_table'):
+            self.get_metadata().drop_all(self.engine_raw, [self.test_table])
+        if hasattr(self, 'executor'):
+            self.executor.execute('drop table if exists empty_table;')
+        logs = self.read_log(self.acra)
+        print(logs)
+        super().tearDown()
+
+    def test_select(self):
+        """verify that after valid SELECT query over transparently encrypted data same config will not be applied
+        for the next query and will be cleared"""
+        encrypted_row = {'nullable': None, 'empty': b'', 'token_i32': random_int32(), 'token_i64': random_int64(),
+                         'token_str': random_str(), 'token_bytes': random_bytes(), 'token_email': random_email()}
+        with self.engine1.begin() as connection:
+            connection.execute(self.test_table.insert(encrypted_row))
+            result = connection.execute(sa.select(self.test_table)).fetchall()
+            self.assertEqual(len(result), 1)
+            for row in result:
+                for k, v in encrypted_row.items():
+                    self.assertEqual(row[k], v)
+            sql = ("select 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h' from empty_table join ("
+                   "select 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h' from empty_table) t on true;")
+            connection.execute(sa.text(sql)).fetchall()
+            logs = self.read_log(self.acra)
+            self.assertIn("can't extract columns from select statement", logs.lower())
+        raw_data = self.engine_raw.execute(sa.select(self.test_table)).fetchall()
+        for row in raw_data:
+            for k, v in encrypted_row.items():
+                # skip empty and nullable
+                if not v:
+                    continue
+                self.assertNotEqual(row[k], v)
+
+    def test_insert_returning(self):
+        """verify that after valid INSERT query with RETURNING over transparently encrypted data same config will not
+        be applied for the next query and will be cleared
+        """
+        if not base.TEST_POSTGRESQL:
+            self.skipTest("MySQL doesn't support returning statement for insert")
```

Comment:
you are right, added via raw sql because current version of sqlalchemy doesn't allow such expressions for mariadb and upgrading to 2.x requires changes and updating tests and should be done in separate PR, imho